From 5bc902451769bfefeda3b20189639516311f7b03 Mon Sep 17 00:00:00 2001
From: David Disseldorp <ddiss@suse.de>
Date: Mon, 8 Oct 2018 14:17:53 +0200
Subject: [PATCH 5/5] target: stash sess_err_stats on Data-Out timeout
References: bsc#1095805
Patch-mainline: v4.20-rc1
Git-commit: 33b3f8ca510a2181574d6dcbd312c2b07dd9f0fa

sess_err_stats are currently filled on NOP ping timeout, but not
Data-Out timeout. Stash details of Data-Out timeouts using a
ISCSI_SESS_ERR_CXN_TIMEOUT value for last_sess_failure_type.

Signed-off-by: David Disseldorp <ddiss@suse.de>

---
 drivers/target/iscsi/iscsi_target_erl1.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/target/iscsi/iscsi_target_erl1.c b/drivers/target/iscsi/iscsi_target_erl1.c
index a77984280107..167c8ebaed28 100644
--- a/drivers/target/iscsi/iscsi_target_erl1.c
+++ b/drivers/target/iscsi/iscsi_target_erl1.c
@@ -1231,6 +1231,7 @@ static void iscsit_handle_dataout_timeout(unsigned long data)
 
 failure:
 	spin_unlock_bh(&cmd->dataout_timeout_lock);
+	iscsit_fill_cxn_timeout_err_stats(sess);
 	iscsit_cause_connection_reinstatement(conn, 0);
 	iscsit_dec_conn_usage_count(conn);
 }
-- 
2.13.7

